src/gluonnlp/vocab/vocab.py

```diff
@@ -307,8 +311,7 @@ def set_embedding(self, *embeddings):
         new_embedding._token_to_idx = self.token_to_idx
         new_embedding._idx_to_token = self.idx_to_token
 
-        new_vec_len = sum(embs.idx_to_vec.shape[1] for embs in embeddings
-                          if embs and embs.idx_to_vec is not None)
+        new_vec_len = sum(embs.idx_to_vec.shape[1] for embs in embeddings)
```

Comment:
@astonzhang removing `if embs and embs.idx_to_vec is not None`. It does not make sense to to `set_embedding` with an embedding that does not have `embs.idx_to_vec` as we do not know the embedding dimensionality. Instead of silently ignoring the error, I added an assertion above.

Consequence of silent ignoring is that `new_vec_len` may be `0` and consequently `vocab.embedding.idx_to_vec.shape == (X, 0)` which is invalid. See also the `test_vocab_set_embedding_with_subword_lookup_only_token_embedding` test below

Reply:
LGTM